### PR TITLE
fix: build should support lib64 dir for OpenSSL3.x

### PR DESCRIPTION
**Pull request by southorange0929**

### CODE REVIEW

File: libsqlite3-sys/build.rs [hunk 1]
```diff
@@ -167,7 +167,14 @@ mod build_bundled {
                         }
                     }
                     Some(openssl_dir) => {
-                        let lib_dir = lib_dir.unwrap_or_else(|| openssl_dir.join("lib"));
+                        let lib_dir = lib_dir.unwrap_or_else(|| {
+                            // OpenSSL 3.0 now puts it's libraries in lib64/ by default,
+                            // check for both it and lib/.
+                            if openssl_dir.join("lib64").exists() {
+                                return openssl_dir.join("lib64");
```
Comment (gwenn) [inline libsqlite3-sys/build.rs:174]: Should depend on `target`, no ?

Comment (southorange0929) [reply libsqlite3-sys/build.rs:174]: I think not, for some architectures, OpenSSL version 3.x is installed under the lib64 directory.

This approach is also in sync with [rust-openssl](https://github.com/sfackler/rust-openssl/blob/master/openssl-sys/build/find_normal.rs#L20).

Comment (gwenn) [reply libsqlite3-sys/build.rs:174]: But `rust-openssl` uses both `lib64` **and** `lib`.

Comment (southorange0929) [reply libsqlite3-sys/build.rs]: Yes, I've now fully aligned its logic with that of openssl.

Comment (gwenn) [reply libsqlite3-sys/build.rs]: Thanks.

**Comment by southorange0929** [comment]
Hi how about to merge this pr? Thanks

**Comment by gwenn** [comment]
Thanks.

**Comment by gwenn** [comment]
libsqlite3-sys v0.29.0 released

**Pull request merged**